### PR TITLE
sasl.oauthbearer.extensions should be optional

### PR DESCRIPTION
**Pull request by lpsinger**
Fixes confluentinc/confluent-kafka-python#1269.

### CODE REVIEW

File: src/rdkafka_conf.c [hunk 1]
```diff
@@ -3709,12 +3709,6 @@ const char *rd_kafka_conf_finalize(rd_kafka_type_t cltype,
                                        "is mandatory when "
                                        "`sasl.oauthbearer.method=oidc` is set";
                         }
-
-                        if (!conf->sasl.oauthbearer.extensions_str) {
```
Comment (edenhill) [inline src/rdkafka_conf.c:3713]: @kirktrue @jliunyu Are extensions indeed optional for OIDC?

Comment (lpsinger) [reply src/rdkafka_conf.c:3713]: My reading of https://datatracker.ietf.org/doc/html/rfc7628#section-3.1 is that they are optional (emphasis is my own):

> Client responses are a GS2 [RFC5801] header followed by **zero or more** key/value pairs, or it **may be empty**.

File: src/rdkafka_conf.c [hunk 2]
```diff
@@ -952,7 +952,7 @@ static const struct rd_kafka_property rd_kafka_properties[] = {
      "is used. If set it to \"oidc\", OAuth/OIDC login method will "
      "be used. "
      "sasl.oauthbearer.client.id, sasl.oauthbearer.client.secret, "
-     "sasl.oauthbearer.scope, sasl.oauthbearer.extensions, "
+     "sasl.oauthbearer.scope, "
```
Comment (jliunyu) [inline src/rdkafka_conf.c:955]: Hi @edenhill , do we need to separate this description for standard AK and CC?

Comment (edenhill) [reply src/rdkafka_conf.c:955]: @jliunyu No I don't think so. The ccloud config requirements should not be expressed here but in ccloud docs&examples.

**Review by edenhill — APPROVED** [review]
LGTM

**Comment by edenhill** [comment]
Thank you @lpsinger !

**Comment by lpsinger** [comment]
Thank you!

**Pull request merged**